cmd/clusterctl/client/yamlprocessor/simple_processor_test.go

```diff
@@ -80,7 +95,82 @@ func TestSimpleProcessor_GetVariables(t *testing.T) {
 			args: args{
 				data: "yaml with ${C:=default}\n${B}\n${A=foobar}",
 			},
-			want: []string{"A", "B", "C"},
+			want: []VariableMetadata{
+				&variableMetadata{name: "A", required: false, defaultValue: "foobar"},
+				&variableMetadata{name: "B", required: true, defaultValue: ""},
+				&variableMetadata{name: "C", required: false, defaultValue: "default"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			p := NewSimpleProcessor()
+			actual, err := p.GetVariables([]byte(tt.args.data))
+			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actual).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestSimpleProcessor_GetVariables_differentTypes(t *testing.T) {
+	type args struct {
+		data string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []VariableMetadata
+		wantErr bool
+	}{
+		{
+			name: "bool Type data test",
+			args: args{
+				data: "yaml with ${A=false}\n${B}\n${C=true}",
+			},
+			want: []VariableMetadata{
+				&variableMetadata{name: "A", required: false, defaultValue: "false"},
+				&variableMetadata{name: "B", required: true, defaultValue: ""},
+				&variableMetadata{name: "C", required: false, defaultValue: "true"},
+			},
+		},
+		{
+			name: "int Type data test",
+			args: args{
+				data: "yaml with ${A=11233333}\n${B}\n${C=8901}",
+			},
+			want: []VariableMetadata{
+				&variableMetadata{name: "A", required: false, defaultValue: "11233333"},
+				&variableMetadata{name: "B", required: true, defaultValue: ""},
+				&variableMetadata{name: "C", required: false, defaultValue: "8901"},
+			},
+		},
+		{
+			name: "string Type data test",
+			args: args{
+				data: "yaml with ${A=simpleString}\n${B=}\n${C=c}",
+			},
```

Comment:
@fabriziopandini i think this is solved.

Reply:
yes, it seems resolved in the latest version